boto3/dynamodb/table.py

```diff
@@ -145,13 +145,13 @@ def _flush(self):
             RequestItems={self._table_name: items_to_send}
         )
         unprocessed_items = response['UnprocessedItems']
-
-        if unprocessed_items and unprocessed_items[self._table_name]:
-            # Any unprocessed_items are immediately added to the
-            # next batch we send.
-            self._items_buffer.extend(unprocessed_items[self._table_name])
+        if unprocessed_items is None:
```

Comment:
The semantics for this change are slightly different than what I'd proposed. This change accounts explicitly for `None`, but we were previously checking for anything falsey. For the sake of robustness the `if not unprocessed_items` may prevent any unintended failures.

Also it looks like you set `unprocessed_items` to an empty list. Ideally we want to be minimize type swapping for the same variable within a function. It obviously happens because we're accounting for falsey types above, but we should aim to avoid it. Otherwise, it's going to make type hints harder for us if they're added in the future. Was there an added benefit with this approach?

Reply:
Ah sorry i think i misunderstood your previous comment about checking for `None`. No added benefit, but thought it made sense considering we were casting to an empty list if `get` returned `None`. I'll update to more closely reflect what you suggested.